app/Http/Controllers/AnnounceController.php

```diff
@@ -121,11 +122,15 @@ public function announce(Request $request, $passkey)
             return response(Bencode::bencode(['failure reason' => 'Missing download']))->withHeaders(['Content-Type' => 'text/plain']);
         }
 
-        // Check Passkey Against Users Table
-        $user = User::with(['group'])->where('passkey', '=', $passkey)->first();
+        // Check Passkey Against Cache or Users Table
+        if (Cache::has("user.{$passkey}")) {
+            $user = Cache::get("user.{$passkey}");
+        } else {
+            $user = User::where('passkey', '=', $passkey)->first();
+        }
```

Comment:
Rather than hit the cache twice (once to check if the value exists, and then again to retrieve it), why not do something like this?

```php
if ($user = Cache::get("user.{$passkey}") === false) {
    $user = User::where('passkey', '=', $passkey)->first();
}
```

Reply:
Maybe I'm wrong but is it not considered bad practice putting a assignment in a condition?

Instead it the `if else` could be shortened with:
```
$user = Cache::get("user.{$passkey}") ? Cache::get("user.{$passkey}") : User::where('passkey', '=', $passkey)->first();
```

Reply:
Or maybe even 
```
$user = Cache::get("user.{$passkey}") ?? User::where('passkey', '=', $passkey)->first();
```

Reply:
I'm not sure it's considered a "bad practice" (though, I am open to any literature that makes a compelling argument for that case), but to your point, the second method you propose is probably better regardless. (The first method calls `Cache::get()` twice, which we want to avoid.) I like it.

Reply:
https://github.com/HDInnovations/UNIT3D-Community-Edition/pull/1117/commits/ea58e49ad24cd10783fbf05e41e0d55a54fa75eb